### ISSUE TITLE
Problems rendering material design lite components

### ISSUE THREAD

**Issue by akshayarora2009** [facebook/react#5044]
So I was trying to implement a simple login page using React and Google's Material design Lite (MDL) framework. I have come across a very strange thing: The MDL's input with floating label doesn't work when rendered using React.

My code is: 

```
var Login = React.createClass({

    render: function(){

        return(

            <div className="mdl-grid">

            <div className="mdl-cell mdl-cell--2-col"></div>

            <div className="mdl-cell mdl-cell--8-col">

                <form>

                    <div className="mdl-textfield mdl-js-textfield mdl-textfield--floating-label">

                        <input className="mdl-textfield__input" type="text" id="sample3" />

                        <label className="mdl-textfield__label" htmlFor="sample3">Text...</label>

                    </div>

                </form>

            </div>

            </div>

        );
    }

});
```

But when I try to use the same thing directly in the HTML, it works fine.
P.S: By "works fine", I mean the animation and colored line that appears below the text box onfocus. 
I think this has something to do with the Synthetic event system implemented by React. I fiddled around with their javascript and found they are doing a bit of stuff with the native "keydown", "onfocus" events, among others.

Similar thing is happening for other elements like an expanding text field. I haven't checked, but I'm sure React + MDL will break at many components of MDL framework. Is this because of the Synthetic event system?

Any way to work around this to get these both working? Or is it safe to assume that React and MDL are incompatible?

**Comment by akshayarora2009** [comment]
Turns out that MDL's js library was trying to mutate some of the components of React, from outside of it. 

All MDL components with the `mdl-js-*` classes have to be "upgraded" within React. After fiddling more with the MDL, I found the real solution to this problem was to to call `componentHandler.upgradeDom();`  from within `componentDidMount()` and/or `componentDidUpdate()`.

So, the problem was not related to React and sorry guys for the noise that may have been caused due to this post on this repository, I'm just a beginner. :sweat_smile:
